### PR TITLE
Improve message retention

### PR DESCRIPTION
**Pull request by johnwatson484**
https://eaflood.atlassian.net/browse/SFI-1509﻿

### CODE REVIEW

File: app/enrichment/validate-values.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 const validateValues = (value, invoiceLines) => {
   const lineValues = invoiceLines.reduce((x, y) => x + y.value, 0)
   if (lineValues !== value) {
-    throw new Error(`Payment request is invalid. Invoice line values (${lineValues}) do not match header (${value})`)
+    const error = new Error(`Payment request is invalid. Invoice line values (${lineValues}) do not match header (${value})`)
```
Comment (steve-dickinson) [inline app/enrichment/validate-values.js:4]: Does this also need a `cause` for consistency?

File: app/enrichment/validate-header.js [hunk 2]
```diff
@@ -3,7 +3,9 @@ const schema = require('./schemas/header')
 const validateHeader = (header) => {
   const validationResult = schema.validate(header)
   if (validationResult.error) {
-    throw new Error(`Header is invalid. ${validationResult.error.message}`)
+    const error = new Error('Header is invalid', { cause: validationResult.error })
+    error.category = 'validation'
```
Comment (steve-dickinson) [inline app/enrichment/validate-header.js:7]: Could this be a lookup to avoid misspelling?

Comment (steve-dickinson) [inline app/enrichment/validate-header.js:7]: Could this be a lookup to avoid misspelling?

Comment (johnwatson484) [reply app/enrichment/validate-header.js]: Agreed and updated

Comment (johnwatson484) [reply app/enrichment/validate-header.js]: Agreed and updated

File: app/enrichment/validate-values.js (hunk 1 above)
Comment (johnwatson484) [reply app/enrichment/validate-values.js:6]: `cause` is for building up a stack trace of errors so if one error causes another you can chain them.  This is the first error thrown in this chain so there is no `cause` to attach.

**Review by steve-dickinson — APPROVED** [review]
LGTM

**Pull request merged**